bijection-core/src/main/scala/com/twitter/bijection/CollectionInjections.scala

```diff
@@ -81,16 +81,22 @@ trait CollectionInjections extends StringInjections {
       }
       override def invert(d: D): Try[C] = {
         val builder = dc()
-        d foreach { b =>
-          val thisB = inj.invert(b)
+        var failed = false
+        val iter = d.toIterator
+        while (iter.hasNext && !failed) {
```

Comment:
I made this do the iterator approach, but the real thing wart remover was complaining about was the use of return